crates/bevy_time/src/time.rs

```diff
@@ -92,57 +123,190 @@ impl Time {
+        self.raw_elapsed_since_startup += raw_delta;
+        self.raw_seconds_since_startup = self.raw_elapsed_since_startup.as_secs_f32();
+        self.raw_seconds_since_startup_f64 = self.raw_elapsed_since_startup.as_secs_f64();
         self.last_update = Some(instant);
     }
 
-    /// The delta between the current tick and last tick as a [`Duration`]
+    /// Returns the [`Instant`] the app was started.
+    #[inline]
+    pub fn startup(&self) -> Instant {
+        self.startup
+    }
+
+    /// Returns the [`Instant`] when [`update`](Self::update) was first called, if it exists.
+    #[inline]
+    pub fn first_update(&self) -> Option<Instant> {
+        self.first_update
+    }
+
+    /// Returns the [`Instant`] when [`update`](Self::update) was last called, if it exists.
+    #[inline]
+    pub fn last_update(&self) -> Option<Instant> {
+        self.last_update
+    }
+
+    /// Returns the rate that time advances relative to raw CPU time, as [`f32`].
+    #[inline]
+    pub fn relative_speed(&self) -> f32 {
+        self.relative_speed as f32
+    }
+
+    /// Returns the rate that time advances relative to raw CPU time, as [`f64`].
+    #[inline]
+    pub fn relative_speed_f64(&self) -> f64 {
+        self.relative_speed
+    }
+
+    /// Sets the rate that time advances relative to raw CPU time, given as [`f32`].
+    ///
+    /// # Panics
+    ///
+    /// Panics if `ratio` is negative or not finite.
+    pub fn set_relative_speed(&mut self, ratio: f32) {
+        assert!(ratio.is_finite(), "tried to go infinitely fast");
+        assert!(ratio.is_sign_positive(), "tried to go back in time");
+        if self.paused {
+            self.stored_relative_speed = Some(ratio as f64);
+        } else {
+            self.relative_speed = ratio as f64;
+        }
+    }
+
+    /// Sets the rate that time advances relative to raw CPU time, given as [`f64`].
+    ///
+    /// # Panics
+    ///
+    /// Panics if `ratio` is negative or not finite.
+    pub fn set_relative_speed_f64(&mut self, ratio: f64) {
+        assert!(ratio.is_finite(), "tried to go infinitely fast");
+        assert!(ratio.is_sign_positive(), "tried to go back in time");
+        if self.paused {
+            self.stored_relative_speed = Some(ratio);
+        } else {
+            self.relative_speed = ratio;
+        }
+    }
+
+    /// Stops time, preventing it from advancing until resumed. Does not affect raw measurements.
+    pub fn pause(&mut self) {
+        if !self.paused {
+            self.stored_relative_speed = Some(self.relative_speed);
+            self.relative_speed = 0.0;
+            self.paused = true;
+        }
+    }
+
+    /// Resumes time if paused.
+    pub fn unpause(&mut self) {
+        if self.paused {
+            // have to pause first to unpause, so something should be stored
+            self.relative_speed = self.stored_relative_speed.take().unwrap();
+            self.paused = false;
+        }
+    }
```

Comment:
The values of `paused` and `stored_relative_speed` are connected : when `paused` is true, `stored_relative_speed` is `Some` and when `paused` is false, `sotred_relative_speed` is `None`.
What about merge them together either in a `Option` or in a dedicated enum ?

Reply:
I made it simpler by moving the check to the getter.